node-manager/src/dlcstorage.rs

```diff
@@ -0,0 +1,365 @@
+            a @ Channel::Accepted(_) | a @ Channel::Signed(_) => {
+                let key = format! {"{DLC_CHANNEL_KEY_PREFIX}{}", a.get_temporary_id().to_hex()};
+                MutinyBrowserStorage::delete(key);
+            }
+            _ => {}
+        };
+
+        let chan_key = format!("{DLC_CHANNEL_KEY_PREFIX}{}", channel.get_id().to_hex());
+        self.set(chan_key, serialized.clone())?;
+
+        if let Some(c) = contract.as_ref() {
+            return self.update_contract(c);
+        } else {
+            Ok(())
+        }
+    }
+
+    fn delete_channel(&mut self, channel_id: &ChannelId) -> Result<(), Error> {
+        let chan_key = format!("{DLC_CHANNEL_KEY_PREFIX}{}", channel_id.to_hex());
+        MutinyBrowserStorage::delete(chan_key);
+        Ok(())
+    }
+
+    fn get_channel(&self, channel_id: &ChannelId) -> Result<Option<Channel>, Error> {
+        let key = format! {"{DLC_CHANNEL_KEY_PREFIX}{}", channel_id.to_hex()};
+        let result: Result<Vec<u8>, MutinyStorageError> = self.get(key);
+        match result {
+            Ok(chan_bytes) => {
+                let channel = deserialize_channel(chan_bytes)
+                    .map_err(|e| Error::StorageError(format!("{}", e)))?;
+                Ok(Some(channel))
+            }
+            Err(_) => Ok(None),
+        }
+    }
+
+    fn get_signed_channels(
+        &self,
+        channel_state: Option<SignedChannelStateType>,
+    ) -> Result<Vec<SignedChannel>, Error> {
+        let (prefix, consume) = if let Some(state) = &channel_state {
+            (
+                vec![
+                    ChannelPrefix::Signed.into(),
+                    SignedChannelPrefix::get_prefix(state),
+                ],
+                None,
+            )
+        } else {
+            (vec![ChannelPrefix::Signed.into()], Some(1))
+        };
+
+        let map: HashMap<String, SignedChannel> = self.get_data_with_prefix(&prefix, consume);
+
+        Ok(map.into_values().collect())
+    }
+
+    fn get_offered_channels(&self) -> Result<Vec<OfferedChannel>, Error> {
+        let map: HashMap<String, OfferedChannel> =
+            self.get_data_with_prefix(&[ChannelPrefix::Offered.into()], None);
+
+        Ok(map.into_values().collect())
+    }
+
+    fn persist_chain_monitor(&mut self, monitor: &ChainMonitor) -> Result<(), Error> {
+        let serialized: Vec<u8> = monitor.serialize()?;
+        self.set(DLC_CHAIN_MONITOR_KEY, serialized)?;
+        Ok(())
+    }
+
+    fn get_chain_monitor(&self) -> Result<Option<ChainMonitor>, Error> {
+        let result: Result<Vec<u8>, MutinyStorageError> = self.get(DLC_CHAIN_MONITOR_KEY);
+
+        match result {
+            Ok(bytes) => {
+                let mon = ChainMonitor::deserialize(&mut Cursor::new(bytes)).map_err(|_| {
+                    Error::StorageError("Failed to decode ChainMonitor".to_string())
+                })?;
+                Ok(Some(mon))
+            }
+            Err(_) => Ok(None),
```

Comment:
Could you match on error type here? To distinguish a problem getting the value vs if a value does not currently exist?